src/sidebar/config/test/build-settings-test.js

```diff
@@ -240,19 +240,21 @@ describe('sidebar/config/fetch-config', () => {
           appType: 'via',
         };
         fakeJsonRpc.call.onFirstCall().resolves({ foo: 'baz' }); // host config
-        const result = await fetchConfig(appConfig, fakeWindow);
+        const result = await buildSettings(appConfig, fakeWindow);
         assert.deepEqual(result.services[0].groups, ['group1', 'group2']);
       });
 
       it("creates a merged config where `groups` is a promise when its initial value is '$rpc:requestGroups'", async () => {
-        const appConfig = {
+        fakeJsonRpc.call.onFirstCall().resolves({
           services: [{ groups: '$rpc:requestGroups' }],
-          appType: 'via',
-        };
-        fakeJsonRpc.call.onFirstCall().resolves({ foo: 'baz' }); // host config
+          foo: 'baz',
+        }); // host config
         fakeJsonRpc.call.onSecondCall().resolves(['group1', 'group2']); // requestGroups
-        const result = await fetchConfig(appConfig, fakeWindow);
-        assert.deepEqual(await result.services[0].groups, ['group1', 'group2']);
+        const result = await buildSettings(/** appConfig */ {}, fakeWindow);
+        const theGroups = await result.services[0].groups;
+
+        assert.deepEqual(theGroups, ['group1', 'group2']);
+
```

Comment:
This test and the one following were adjusted because I believe the previous behavior was not actually what we intended. `Service` values are expected to be present on `HostConfig`

https://github.com/hypothesis/client/blob/7d7ae510a5817aba7c5e0a6dba27ac6f2d9acf57/src/types/config.js#L81

But previous implementation and tests would also look for `services` on `SidebarConfig` (`appConfig` in the changed test) — I don't think that's a valid location for them, in the very least not for RPC-fetched host configuration (where all services information should come from that RPC-returned configuration object).